client/src/components/molecules/CategoryMatcherAdder/CategoryMatcherAdder.tsx

```diff
@@ -35,7 +35,7 @@ export const CategoryMatcherAdder = ({
     }
 
     form.setFieldsValue({
-      regexExpression: `${record.description}`,
+      regexExpression: `${record.description.replaceAll(/(?<![\[\]])\*(?![\[\]])/g, "[*]")}`,
```

Comment:
let's extract `parseSpecialChar` to some util file and use it in both `CategoryMatcherAdder` and `useGetRegexMatches`. Also let's include some basic tests